internal/pkg/jwt/jwt.go

```diff
@@ -74,15 +77,49 @@ func (sc *DefaultJWTConsumer) getPublicKey(kid string) (*rsa.PublicKey, error) {
 		//ctx.Logger().Error("op", "JWTConsumer.getKeyData", "stage", "client.DoRequest", "status", status)
 		return nil, fmt.Errorf("Invalid status code: %d\n", resp.StatusCode)
 	}
-	pub, err = jwt.ParseRSAPublicKeyFromPEM(bs)
-	if nil != err {
-		//ctx.Logger().Error("op", "JWTConsumer.getKeyData", "stage", "decode", "err", err)
-		return nil, err
+	if strings.Contains(sc.publicKeyEndpoint, "v2") {
```

Comment:
This logic seems to be unreliable.  Can we do something else?  For example, use the actual token content to determine the behavior?

Reply:
I feel this is the easiest and best performance way. We can decode the token and find the version in the token and based on it to use different way. We can also try to pass the value in v1 format and then if it fails, use the v2 format. I feel check string v2 will save all of these.